src/dawn/Optimizer/PassSetNonTempCaches.cpp

```diff
@@ -0,0 +1,325 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassSetNonTempCaches.h"
+#include "dawn/Optimizer/Cache.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/PassDataLocalityMetric.h"
+#include "dawn/Optimizer/PassSetCaches.h"
+#include "dawn/Optimizer/StatementAccessesPair.h"
+#include "dawn/Optimizer/StencilInstantiation.h"
+#include "dawn/SIR/ASTExpr.h"
+#include "dawn/Support/Unreachable.h"
+#include <iostream>
+#include <set>
+#include <vector>
+
+namespace dawn {
+
+struct accessIDaccessMetric {
```

Comment:
Classes should start with a capital letter.

Reply:
sure